Saule/Resources/DefaultApiResourceProviderFactory.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Net.Http;
+
+namespace Saule.Resources
+{
+    /// <summary>
+    /// Default provider factory that resolves attributes based on attributes and HttpRequestMessage
+    /// </summary>
+    public class DefaultApiResourceProviderFactory : IApiResourceProviderFactory
+    {
+        /// <inheritdoc/>
+        public IApiResourceProvider Create(HttpRequestMessage request)
+        {
+            object resource;
+            if (request.Properties.TryGetValue(Constants.PropertyNames.ResourceDescriptor, out resource) && resource is ApiResource)
+            {
+                var apiResource = (ApiResource)resource;
+                return new DefaultApiResourceProvider(apiResource);
+            }
+
+            // if no resource was specified then we return empty provider that will always return null
+            return new DefaultApiResourceProvider(null);
```

Comment:
Does this match the current Saule behavior? I think it makes more sense to throw a nice error message that says what the problem is, but maybe this is not the spot for that?

Reply:
yeah, i described it above but there are some flows like `[HttpDelete]` where resource might be missing and `ApiResource` might be null in that case. and it just won't serialize anything as endpoint returns `void` that would be translated to Http 204 NoContent. if we would force `ApiResource` to always be present, then it might break some existing functionality